### PR TITLE
pokemonlist

### PR DESCRIPTION
**Pull request by FlorianNova**

### CODE REVIEW

File: styles/PokemonCard.styles.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  @media (max-width: 576px) {
+    width: 100%; /* Ändert die Breite der Karten auf noch kleineren Bildschirmen */
+    margin: 0;
+    margin-bottom: 0;
+    margin-top: 0;
+    padding: 0;
+    padding-top: 0;
+  }
```
Comment (Roland-Hufnagel) [inline styles/PokemonCard.styles.js:49]: This mediaquery is not needed since it does not change anything

File: pages/index.js [hunk 2]
```diff
@@ -1,11 +1,26 @@
-import Heading from "../components/Heading";
-import SubHeading from "../components/SubHeading";
+import React from 'react';
+import Head from 'next/head';
+import PokemonList from '../components/PokemonList';
+import * as Styled from '../styles/PokemonCard.styles';
 
-export default function Home() {
+const Home = () => {
```
Comment (Roland-Hufnagel) [inline pages/index.js:6]: Why did you change this?

File: styles/PokemonCard.styles.js (hunk 1 above)
Comment (FlorianNova) [reply styles/PokemonCard.styles.js:49]: That's right, currently it doesn't do anything, but I deliberately didn't delete some things because I might need them again / want to change them, but I can also change them to comments if I have to.

File: pages/index.js [hunk 3]
```diff
@@ -1,11 +1,26 @@
-import Heading from "../components/Heading";
-import SubHeading from "../components/SubHeading";
+import React from 'react';
```
Comment (Roland-Hufnagel) [inline pages/index.js:1]: I think you don't need to import React since this is a react-project.

File: pages/index.js [hunk 4]
```diff
@@ -1,11 +1,26 @@
...
+import * as Styled from '../styles/PokemonCard.styles';
 
-export default function Home() {
+const Home = () => {
   return (
-    <main>
-      <Heading>🏄‍♂️🏄‍♀️🏄 Capstone Template 🏄🏄‍♀️🏄‍♂️</Heading>
-      <SubHeading>Happy hacking!</SubHeading>
-    </main>
+    <div>
```
Comment (Roland-Hufnagel) [inline pages/index.js:8]: Do you need that div here? It would be a cleaner solution to use a fragment <></>

File: pages/index.js [hunk 5]
```diff
@@ -1,11 +1,26 @@
...
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <main>
+        {/* Hier wird der Titel "Pokemon Card List" mittig zentriert */}
+        <Styled.Title>Pokémon Card List</Styled.Title>
+        <PokemonList />
+      </main>
+
+      <footer>{/* hier ggf. den Footer hinzufügen */}</footer>
```
Comment (Roland-Hufnagel) [inline pages/index.js:21]: Since you don't have a footer (and it is not part of your US) remove the code here.

File: pages/index.js [hunk 6]
```diff
@@ -1,11 +1,26 @@
...
-    </main>
+    <div>
+      <Head>
+        <title>PokéTeam Web App - Pokémon Card List</title>
+        <meta name="description" content="Pokémon Card List" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <main>
+        {/* Hier wird der Titel "Pokemon Card List" mittig zentriert */}
```
Comment (Roland-Hufnagel) [inline pages/index.js:16]: it is not a good practice to leave comments in the code. If you think this one  is really important for you, it is okay but try to remove as much comments as possible.

File: components/PokemonList.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
```
Comment (Roland-Hufnagel) [inline components/PokemonList.js:1]: here the same: No need to import React

File: components/PokemonList.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  useEffect(() => {
+    const fetchPokemon = async () => {
+      const data = await fetchPokemonData();
+      setPokemonData(data);
+    };
+    fetchPokemon();
+  }, []);
+
+  return (
+    <Styled.GridWrapper>
```
Comment (Roland-Hufnagel) [inline components/PokemonList.js:18]: You want to render a LIST of pokemoncards here. So please use a ``<ul>`` Element here instead of a ``<div>``

File: components/PokemonList.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      {pokemonData.map((pokemon, index) => (
+        <PokemonCard
```
Comment (Roland-Hufnagel) [inline components/PokemonList.js:20]: And use semantic html here again. Your PokemonCard Component is a ``<div>`` element. Please use a ``<li>`` Element here

File: components/PokemonList.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+          key={index}
```
Comment (Roland-Hufnagel) [inline components/PokemonList.js:21]: I strongly recommend to not use the index for this purpose. I think your pokemon comes with an 'id'. This should be used instead.

File: styles/PokemonCard.styles.js (hunk 1 above)
Comment (FlorianNova) [reply styles/PokemonCard.styles.js:49]: I've now updated the code a bit as mentioned. Should I push it again so you can have a quick look, or should i maybe prefer to wait until you are ready?

File: components/PokemonList.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+          name={pokemon.name}
+          number={index + 1}
+          imageUrl={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${
+            index + 1
+          }.png`}
+          types={pokemon.types || []} // Sicherstellen, dass types ein Array ist oder ein leeres Array verwenden für zukünftiges von Pokemon Typen
```
Comment (Roland-Hufnagel) [inline components/PokemonList.js:27]: Please remove all code, that is not part of the current (and previous) stories.

File: components/PokemonCard.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import * as Styled from '../styles/PokemonCard.styles';
+
+const PokemonCard = ({ name, number, imageUrl, types }) => {
```
Comment (Roland-Hufnagel) [inline components/PokemonCard.js:3]: Also here remove types if it is not part of the current US

File: components/PokemonCard.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+      </Styled.PokemonImage>
+      <Styled.PokemonNumber>{`#${number}`}</Styled.PokemonNumber>
+      <Styled.PokemonName>
+        {name.charAt(0).toUpperCase() + name.slice(1)}
+      </Styled.PokemonName>
+      <Styled.PokemonTypes>
+        {types.map((type, index) => (
+          <Styled.PokemonType key={index}>{type}</Styled.PokemonType>
+        ))}
+      </Styled.PokemonTypes>
```
Comment (Roland-Hufnagel) [inline components/PokemonCard.js:17]: Please remove since it is not part of this US

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hey Florian. Thank you to have a deep dive into your code and into the Poke API 😀👍.
Your first PR looks very good and it works fine. Although I request some changes. Please have a look at my comments. 
And one thing all above: 
It's not good practice to have a 'styles' folder, where you collect all styled Components. It's the idea of react to have all things nearby. This means, you should define  the Styled Components, that you use in a component inside that component. Or even if you have a component folder like 'PokemonCard' that could contain an 'index.js' where you write the component and an 'styles.js' where you define the corresponding Styled Components. Clear? If you have questions feel free to ask me.

All in all it is a very good first PR! 💪

File: pages/index.js (hunk 2 above)
Comment (FlorianNova) [reply pages/index.js:6]: changed

File: pages/index.js (hunk 3 above)
Comment (FlorianNova) [reply pages/index.js:1]: changed

File: pages/index.js (hunk 4 above)
Comment (FlorianNova) [reply pages/index.js:8]: changed

File: pages/index.js (hunk 5 above)
Comment (FlorianNova) [reply pages/index.js:21]: changed

File: pages/index.js (hunk 6 above)
Comment (FlorianNova) [reply pages/index.js:16]: changed

File: components/PokemonList.js (hunk 7 above)
Comment (FlorianNova) [reply components/PokemonList.js:1]: changed

File: components/PokemonList.js (hunk 8 above)
Comment (FlorianNova) [reply components/PokemonList.js:18]: since I don't want to render a real list, but want the appearance of a "list", this is currently not necessary.

File: components/PokemonList.js (hunk 9 above)
Comment (FlorianNova) [reply components/PokemonList.js:20]: since I don't want to render a real list, but want the appearance of a "list", this is currently not necessary too.
